### PR TITLE
Routes, Controllers and Views in the Blog App

### PR DESCRIPTION
**Pull request by ehmaddd**
The following features are added in this Pull Request : 

-  For each URL the followings are created :
  * A route.
  * An action in the correct controller.
  * A view file.

-  View files include only placeholders written in plain HTML

-  Added code to handle only the 4 required URLs

- After running the **rails routes** command in theconsole, you can see the following output :

image

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
+  end
+end
```
Comment (TedLivist) [inline app/controllers/users_controller.rb]: - [x] The Users controller should contain the `index` method.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<h1>User Details</h1>
```
Comment (TedLivist) [inline app/views/users/show.html.erb:1]: - [x] Kindly create the index page for the users `index` action

File: config/routes.rb [hunk 3]
```diff
@@ -2,5 +2,10 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  # root 'home#index'
+  get '/users', to: 'users#index', as: :users
+  get '/users/:id', to: 'users#show', as: :user
+  get '/users/:user_id/posts', to: 'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
+
```
Comment (TedLivist) [inline config/routes.rb]: - [x] **[OPTIONAL]** I think we can define these routes in a more concise manner using the `resources` helper.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ehmaddd,

Good job so far!
✔️ Implementation of nested routes
✔️ Creation of controller actions
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 3 above)
Comment (ehmaddd) [reply config/routes.rb]: Changed as instructed

File: app/views/users/show.html.erb (hunk 2 above)
Comment (ehmaddd) [reply app/views/users/show.html.erb:1]: Created index file as per instructions

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (ehmaddd) [reply app/controllers/users_controller.rb]: Added the method as instructed

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @ehmaddd,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**